library/X509/CertificateUtils.php

```diff
@@ -123,15 +123,21 @@ public static function duration($seconds)
      *
      * If the given DN contains a CN, the CN is returned. Else, the DN is returned as string.
      *
-     * @param   array   $dn
+     * @param array $certInfo
+     * @param bool $isIssuer
      *
      * @return  string  The CN if it exists or the full DN as string
      */
-    private static function shortNameFromDN(array $dn)
+    private static function shortNameFromDN(array $certInfo, bool $isIssuer = true): string
     {
-        if (isset($dn['CN'])) {
-            $cn = (array) $dn['CN'];
-            return $cn[0];
+        $dn = $isIssuer ? $certInfo['issuer'] : $certInfo['subject'];
+        if (isset($dn['CN']) || isset($certInfo['extensions']['subjectAltName'])) {
```

Comment:
I would let `insertSANs()` expect the generator from `splitSANs()`. This way you don't have duplicate code and don't have to call `iterator_to_array()`, e.g.

```
$sans = splitSANs();

$db->insert(..., [..., 'subject' => shortNameFromDN($certInfo['subject']) ?? $sans->current()[1] ?? '']);

insertSANs(..., $sans);
```